core/stream/convert.go

```diff
@@ -388,9 +409,29 @@ func ftou(count int, dst, src buf) error {
 		}
 	case srcIsF32:
 		scale := float32(dstMask)
+		min, max := float32(math.MaxFloat32), float32(0)
+		for i := 0; i < count; i++ {
+			f := math.Float32frombits(uint32(ss2.Read(32)))
+			if f < min {
+				min = f
+			}
+			if f > max {
+				max = f
+			}
+			ss2.ReadPos += src.stride - 32
+		}
 		for i := 0; i < count; i++ {
 			f := math.Float32frombits(uint32(ss.Read(32)))
 			if norm {
+				if src.component.Channel == Channel_Depth {
+					f = (f - min) * (1 / (max - min))
+				} else if min < 0 {
+					if f <= 0 {
+						f = (f - min) * (0.5 / -min)
+					} else {
+						f = f*(0.5/max) + 0.5
```

Comment:
possible division by zero

Reply:
Nope. If max were zero, we would never have a positive f

Reply:
Stability concern?